packages/babel-types/src/definitions/experimental.js

```diff
@@ -258,3 +258,25 @@ defineType("BigIntLiteral", {
   },
   aliases: ["Expression", "Pureish", "Literal", "Immutable"],
 });
+
+defineType("SliceExpression", {
+  visitor: ["object", "begin", "end", "step"],
+  aliases: ["Expression", "LVal"],
+  fields: {
+    object: {
+      validate: assertNodeType("Expression"),
+    },
+    lower: {
```

Comment:
May consider rename `lower/upper` to `startIndex/endIndex` given that step can be negative when `lower` is not necessarily lower than `upper`.
```js
const arr = [0, 1, 2]
arr[2:1:-1] // [2]
```

Reply:
Is this actually valid?

Reply:
> If the step argument is negative, then the object is traversed in reverse.

Yes if you mean the negative step.